docs/rackhd/npm_based_installation.rst

```diff
@@ -0,0 +1,242 @@
+
+   .. code::
+
+    curl --silent https://deb.nodesource.com/gpgkey/nodesource.gpg.key | sudo apt-key add -
+    VERSION=node_4.x
+    DISTRO="$(lsb_release -s -c)"
+    echo "deb https://deb.nodesource.com/$VERSION $DISTRO main" | sudo tee /etc/apt/sources.list.d/nodesource.list
+    echo "deb-src https://deb.nodesource.com/$VERSION $DISTRO main" | sudo tee -a /etc/apt/sources.list.d/nodesource.list
+
+    sudo apt-get update
+    sudo apt-get install nodejs
+
+* **Ensure Node.js is at version 4.x**
+
+   For Example:
+
+   .. code::
+
+    $ node -v
+    v4.4.5
+
+
+2. **Dependencies**
+
+* **Install dependency pakcages**
+
+   .. code::
+
+    sudo apt-get install rabbitmq-server
+    sudo apt-get install mongodb
+    sudo apt-get install snmp
+    sudo apt-get install ipmitool
+    
+    sudo apt-get install git
+    sudo apt-get install unzip
+    sudo apt-get install ansible
+    sudo apt-get install apt-mirror
+    sudo apt-get install amtterm
+
+    sudo apt-get install isc-dhcp-server
+
+
+####
+
+Install & Configure RackHD
+^^^^
+
+1. **Install RackHD NPM package**
+
+*
+   .. code::
+
+     for service in $(echo "on-dhcp-proxy on-http on-tftp on-syslog on-taskgraph");
+     do npm install $service;
+     done
+
+
+2. **Basic RackHD Configuration**
+
+
+* **DHCP**
+
+   Update /etc/dhcp/dhcpd.conf per your network configuration
+ 
+   .. code::
+
+    # RackHD added lines
+    deny duplicates;
+
+    ignore-client-uids true;
+
+    subnet 172.31.128.0 netmask 255.255.240.0 {
+      range 172.31.128.2 172.31.143.254;
+      # Use this option to signal to the PXE client that we are doing proxy DHCP
+      option vendor-class-identifier "PXEClient";
+    }
+
+
+#######
+
+* **RACKHD APPLICATIONS**
+
+   Create the required file /opt/monorail/config.json , you can use the demonstration configuration file at https://github.com/RackHD/RackHD/blob/master/packer/ansible/roles/monorail/files/config.json as a reference.
+
+#######
+
+* **RACKHD BINARY SUPPORT FILES**
+
+   Downloaded binary files from bintray.com/rackhd/binary and placed them.
+
+   .. code::
+
+    #!/bin/bash
+
+    mkdir -p node_modules/on-tftp/static/tftp
+    cd node_modules/on-tftp/static/tftp
+
+    for file in $(echo "\
+    monorail.ipxe \
+    monorail-undionly.kpxe \
+    monorail-efi64-snponly.efi \
+    monorail-efi32-snponly.efi");do
+    wget "https://dl.bintray.com/rackhd/binary/ipxe/$file"
+    done
+
+    cd -
+
+    mkdir -p node_modules/on-http/static/http/common
+    cd node_modules/on-http/static/http/common
+
+    for file in $(echo "\
+    base.trusty.3.16.0-25-generic.squashfs.img \
+    discovery.overlay.cpio.gz \
+    initrd.img-3.16.0-25-generic \
+    vmlinuz-3.16.0-25-generic");do
+    wget "https://dl.bintray.com/rackhd/binary/builds/$file"
+    done
+
+    cd -
+
+3. **Start RackHD**
+
+   Start the 5 components of RackHD with pm2.
```

Comment:
What's more, don't mix use the word `component` and `service`, sometime you use `component`, sometime `service`. Keep using the same word will be better